filters/filter.go

```diff
@@ -465,6 +506,46 @@ func (f *singleCharSetFilter) matches(val string) (string, bool) {
 	return val[1:], match
 }
 
+// multiCharRangeFilter is a filter that performs matches against multiple sets of chars
```

Comment:
Perhaps this should be called multiCharListFilter or multiCharSequenceFilter? It's not matching against range per se